docs/spec/spec.md

```diff
@@ -267,6 +273,189 @@ The GraphQL engine acts as the main processing unit in the Ballerina GraphQL pac
+
+```ballerina
+# Provides a set of configurations for controlling the behavior of the GraphQL client when communicating with 
+# the GraphQL server that operates over HTTP.
+#
+# + http1Settings - Configurations related to HTTP/1.1 protocol
+# + timeout - The maximum time to wait (in seconds) for a response before closing the connection
+# + forwarded - The choice of setting `forwarded`/`x-forwarded` header
+# + followRedirects - Configurations associated with Redirection
+# + poolConfig - Configurations associated with request pooling
+# + cache - HTTP caching related configurations
+# + compression - Specifies the way of handling compression (`accept-encoding`) header
+# + auth - Configurations related to client authentication
+# + circuitBreaker - Configurations associated with the behavior of the Circuit Breaker
+# + retryConfig - Configurations associated with retrying
+# + cookieConfig - Configurations associated with cookies
+# + responseLimits - Configurations associated with inbound response size limits
+# + secureSocket - SSL/TLS-related options
+# + proxy - Proxy server related options
+# + validation - Enables the inbound payload validation functionality which is provided by the constraint package. Enabled by default
+public type ClientConfiguration record {|
+    ClientHttp1Settings http1Settings = {};
+    decimal timeout = 60;
+    string forwarded = "disable";
+    FollowRedirects? followRedirects = ();
+    PoolConfiguration? poolConfig = ();
+    CacheConfig cache = {};
+    Compression compression = COMPRESSION_AUTO;
+    ClientAuthConfig? auth = ();
+    CircuitBreakerConfig? circuitBreaker = ();
+    RetryConfig? retryConfig = ();
+    CookieConfig? cookieConfig = ();
+    ResponseLimitConfigs responseLimits = {};
+    ClientSecureSocket? secureSocket = ();
+    ProxyConfig? proxy = ();
+    boolean validation = true;
+|};
+```
+
+#### 2.5.2 Initializing the Client
+
+The `graphql:Client` init method requires a valid URL and optional configuration to initialize the client. 
+
+```ballerina
+graphql:Client graphqlClient = check new (“http://localhost:9091/graphql”, {timeout: 10});
+```
+#### 2.5.3 Executing Operations
+
+The graphql client provides `execute` API to execute graphql query and mutation operations. The `execute` method of `graphql:Client` takes a GraphQL document as the required argument and sends a request to the specified backend URL seeking for a response. Further, the execute method could take the following optional arguments.
+
+- `variables` - A map containing the GraphQL variables. All the variables that may be required by the graphql document can be set via this variables argument.
+
+- `operationName` - The GraphQL operation name. If the document has more than one operation, then each operation must have a name. Single GraphQL request can only execute one operation; the operation name must be set if the document has more than one operation.
```

Comment:
What happens when the name is not provided and there are multiple operations? Will that return an error from the client-side or will it be sent to the server? Shall we check and update the behavior here?

Reply:
I think we can improve the behavior here. We can use the existing GraphQL parser to parse and validate the document before sending it out. Shall we create an issue for this?

Reply:
But the syntax validation will be duplicated in both places (client and server). 
In most cases there wont be any error in the query (or once syntactically invalid query is corrected). So it will take more time to execute a valid query.
Having syntax validation in client side seems to be redundant IMO. WDYT ?

Reply:
Point! Let's keep it as it is then.